presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/LongDecimalWithOverflowState.java

```diff
@@ -15,14 +15,17 @@
 
 import com.facebook.presto.spi.function.AccumulatorState;
 import com.facebook.presto.spi.function.AccumulatorStateMetadata;
+import io.airlift.slice.Slice;
 
-import java.math.BigInteger;
-
-@AccumulatorStateMetadata(stateFactoryClass = BigIntegerStateFactory.class, stateSerializerClass = BigIntegerStateSerializer.class)
-public interface BigIntegerState
+@AccumulatorStateMetadata(stateFactoryClass = LongDecimalWithOverflowStateFactory.class, stateSerializerClass = LongDecimalWithOverflowStateSerializer.class)
+public interface LongDecimalWithOverflowState
         extends AccumulatorState
 {
-    BigInteger getBigInteger();
+    Slice getLongDecimal();
+
+    void setLongDecimal(Slice unscaledDecimal);
```

Comment:
shouldn't you set both long decimal and overflow at the same time? It's strange that you can set them separately.